Travis failed:

```
Error response from daemon: Cannot start container 51ea09227a4c41b3a639040daac2124f8ed76f5e0d61d6838799de1919dfbcb0: Cannot link to a non running container: /cocky_mahavira AS /insane_cray/minio

make: *** [test] Error 1

The command "make test" exited with 2.
```

--- a/contrib/ci/test.sh
+++ b/contrib/ci/test.sh
@@ -32,8 +32,10 @@ echo "1234567890123456789012345678901234567890" > $CURRENT_DIR/tmp/aws-admin/acc
 # create fake AWS credentials for minio user credentials
 mkdir -p $CURRENT_DIR/tmp/aws-user
 # needs to be 20 characters long
+echo "12345678901234567890" > $CURRENT_DIR/tmp/aws-user/accesskey
 echo "12345678901234567890" > $CURRENT_DIR/tmp/aws-user/access-key-id
 # needs to be 40 characters long
+echo "1234567890123456789012345678901234567890" > $CURRENT_DIR/tmp/aws-user/secretkey
 echo "1234567890123456789012345678901234567890" > $CURRENT_DIR/tmp/aws-user/access-secret-key
 
 puts-step "creating fake kubernetes service account token"
@@ -47,7 +49,7 @@ echo "cert" > $CURRENT_DIR/tmp/k8s/ca.crt
 MINIO_JOB=$(docker run -dv $CURRENT_DIR/tmp/aws-admin:/var/run/secrets/deis/minio/admin -v $CURRENT_DIR/tmp/aws-user:/var/run/secrets/deis/minio/user -v $CURRENT_DIR/tmp/k8s:/var/run/secrets/kubernetes.io/serviceaccount quay.io/deisci/minio:canary boot server /home/minio/)
 
 # boot postgres, linking the minio container and setting DEIS_MINIO_SERVICE_HOST and DEIS_MINIO_SERVICE_PORT
-PG_JOB=$(docker run -d --link $MINIO_JOB:minio -e BACKUP_FREQUENCY=1s -e DEIS_MINIO_SERVICE_HOST=minio -e DEIS_MINIO_SERVICE_PORT=9000 -v $CURRENT_DIR/tmp/creds:/var/run/secrets/deis/database/creds -v $CURRENT_DIR/tmp/aws-user:/etc/wal-e.d/env $1)
+PG_JOB=$(docker run -d --link $MINIO_JOB:minio -e BACKUP_FREQUENCY=1s -e DATABASE_STORAGE=minio -e DEIS_MINIO_SERVICE_HOST=minio -e DEIS_MINIO_SERVICE_PORT=9000 -v $CURRENT_DIR/tmp/creds:/var/run/secrets/deis/database/creds -v $CURRENT_DIR/tmp/aws-user:/var/run/secrets/deis/objectstore/creds $1)
 
 # wait for postgres to boot
 puts-step "sleeping for 90s while postgres is booting..."

--- a/rootfs/Dockerfile
+++ b/rootfs/Dockerfile
@@ -58,7 +58,7 @@ RUN curl -sSL https://raw.githubusercontent.com/pypa/pip/7.1.2/contrib/get-pip.p
 
 # install deis/wal-e from the tip of the boto3-upgrade branch
 # NOTE (bacongobbler): we use the commit here so Docker's cache will bust when we make changes upstream
-RUN pip install --disable-pip-version-check --no-cache-dir git+https://github.com/kmala/wal-e.git@3c1729302db6a38ae32709da24d91a5caa264499
+RUN pip install --disable-pip-version-check --no-cache-dir git+https://github.com/deis/wal-e.git@0eee12665149d8389cda9575b737359d066fa663
 
 # install python port of daemontools
 RUN pip install --disable-pip-version-check --no-cache-dir envdir

--- a/rootfs/Dockerfile
+++ b/rootfs/Dockerfile
@@ -58,7 +58,7 @@ RUN curl -sSL https://raw.githubusercontent.com/pypa/pip/7.1.2/contrib/get-pip.p
 
 # install deis/wal-e from the tip of the boto3-upgrade branch
 # NOTE (bacongobbler): we use the commit here so Docker's cache will bust when we make changes upstream
-RUN pip install --disable-pip-version-check --no-cache-dir git+https://github.com/kmala/wal-e.git@5c869f9ecc73b2a73a4b87b41b7b44be086b1e4a
+RUN pip install --disable-pip-version-check --no-cache-dir git+https://github.com/kmala/wal-e.git@3c1729302db6a38ae32709da24d91a5caa264499
 
 # install python port of daemontools
 RUN pip install --disable-pip-version-check --no-cache-dir envdir

--- a/rootfs/bin/create_bucket
+++ b/rootfs/bin/create_bucket
@@ -11,7 +11,7 @@ from botocore.client import Config
 from oauth2client.service_account import ServiceAccountCredentials
 from gcloud.storage.client import Client
 from gcloud import exceptions
-from azure.storage.blob import BlobService
+from azure.storage.blob import BlockBlobService
 
 bucket_name = os.getenv('BUCKET_NAME')
 
@@ -49,9 +49,9 @@ elif os.getenv('DATABASE_STORAGE') == "gcs":
     	client.create_bucket(bucket_name)
 
 elif os.getenv('DATABASE_STORAGE') == "azure":
-  blob_service = BlobService(account_name=os.getenv('WABS_ACCOUNT_NAME'), account_key=os.getenv('WABS_ACCESS_KEY'))
-  #It doesn't throw an exception if the container exists by default(https://github.com/Azure/azure-storage-python/blob/v0.20.3/azure/storage/blob/blobservice.py#L298).
-  blob_service.create_container(bucket_name)
+  block_blob_service = BlockBlobService(account_name=os.getenv('WABS_ACCOUNT_NAME'), account_key=os.getenv('WABS_ACCESS_KEY'))
+  #It doesn't throw an exception if the container exists by default(https://github.com/Azure/azure-storage-python/blob/master/azure/storage/blob/baseblobservice.py#L504).
+  block_blob_service.create_container(bucket_name)
 
 else :
   conn = boto3.resource('s3', endpoint_url=os.getenv('S3_URL'), config=Config(signature_version='s3v4'))

